### DISCUSSION TITLE
More claim protection

### DISCUSSION THREAD

**Discussion by MrBuushy** [GriefPrevention/GriefPrevention#440] [Ideas]
### Can you please make a claim command just for administrative claims where you can choose if you want these on and off: Mob-Spawning and Mob-Entering

**_Permission Nodes:_** Whatever you want maybe like griefprevention.admin.mobspawning and griefprevention.admin.mobentering

**_Commands:_**        /ClaimMobSpawning on/off         and          /ClaimMobEntering on/off

Because I usually use claims to protect spawn and worldguard to stop mobs spawning (not entering) and it's annoying because first, worldgaurd is stupidly hard to use and seconds, I like griefprevention and want to use that only, so could please add these and also worldguard does not have a mob-entering flag.

**Comment by bigpresh** [comment]
I suspect these would be better off handled by a plugin, but shouldn't be too difficult.  I imagine checking mob entering could be reasonably inefficient, though.

**Comment by MrBuushy** [comment]
> I suspect these would be better off handled by a plugin, but shouldn't be too difficult. I imagine checking mob entering could be reasonably inefficient, though.

why? is there not a way it can scan mobs around the area, and if the mob gets too close it gets kill/removed?, i remember this on a server that shut down ages ago but i think that was a custom addon plugin to this plugin so i wondered if you could add it or not

**Comment by bigpresh** [comment]
I was thinking of something that receives EntityMoveEvent events and checks if that entity is a hostile mob and the position it's moving to is within a claim with such a flag set - but yes, it might indeed be more efficient to just iterate all claims every few seconds or so and for each one with the flag set, look for any hostile mobs whose position is within that claim and remove them if found.

I still think it's better suited as a plugin rather than core functionality, though.

In fact, I'm not sure but what you're looking for might already be offered by the Flags plugin: https://dev.bukkit.org/projects/flags

It has a "Creature Spawn Module" which controls spawning, and a "Border Patrol Module" - although the latter seems to be focussed on players, it may well be easy enough for someone to use it as the basis for a module that stops mobs entering/exiting claims.

**Comment by RoboMWM** [comment]
This is already available in GriefPrevention flags. There are a couple maintained forks out there, the most updated one doesn't seem to have any released builds though... you can compile with maven however.

**Comment by MrBuushy** [comment]
> This is already available in GriefPrevention flags. There are a couple maintained forks out there, the most updated one doesn't seem to have any released builds though... you can compile with maven however.

mob-entering is not a flag

**Comment by MrBuushy** [comment]
I think it should be a core function because players can obviously stop mob spawning but can't stop mobs entering and some people don't know about plugin addons

> I was thinking of something that receives EntityMoveEvent events and checks if that entity is a hostile mob and the position it's moving to is within a claim with such a flag set - but yes, it might indeed be more efficient to just iterate all claims every few seconds or so and for each one with the flag set, look for any hostile mobs whose position is within that claim and remove them if found.
> 
> I still think it's better suited as a plugin rather than core functionality, though.
> 
> In fact, I'm not sure but what you're looking for might already be offered by the Flags plugin: https://dev.bukkit.org/projects/flags
> 
> It has a "Creature Spawn Module" which controls spawning, and a "Border Patrol Module" - although the latter seems to be focussed on players, it may well be easy enough for someone to use it as the basis for a module that stops mobs entering/exiting claims.

I think it should be a core function because players can obviously stop mob spawning but can't stop mobs entering and some people don't know about plugin addons

**Comment by RoboMWM** [comment]
> I think it should be a core function 

Don't see how this should be a core function at all.

**Comment by MrBuushy** [comment]
> > I think it should be a core function
> 
> Don't see how this should be a core function at all.

because players can obviously stop mob spawning but can't stop mobs entering and some people don't know about plugin addons

**Comment by RoboMWM** [comment]
> because players can obviously stop mob spawning 

Via an addon though. Unless you mean using light and non-spawnable blocks like slabs; in which case, then it's also possible to prevent entry via vanilla methods as well (building walls, fence, etc.) 

Again, still don't see how this fits in the core theme of the plugin. The most mobs can do is damage animals and players, provided they can enter said area with animals. AFAIK the theme of griefprevention is to prevent intentional grief by other players. (AI does not have such a malicious concept or choice.)

**Comment by MrBuushy** [comment]
@RoboMWM Just wondering, is this going to be added?
